src/mongo/db/curop.h

```diff
@@ -305,6 +306,13 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
+            const int64_t RATE_LIMIT_MULTIPLIER = 1LL << 52;
+            static_assert( RATE_LIMIT_MAX * RATE_LIMIT_MULTIPLIER <= std::numeric_limits<int64_t>::max(),
+                          "product of RATE_LIMIT_MAX and RATE_LIMIT_MULTIPLIER should not exceed int64_t range");
+            return _prng.nextInt64(RATE_LIMIT_MULTIPLIER) * serverGlobalParams.rateLimit < RATE_LIMIT_MULTIPLIER;
```

Comment:
Well, you are right. Your version also should work. I only have some doubts on the quality of generated sequences. So if you don't mind I would prefer to leave my version.